tests/DocumentTest.php

```diff
@@ -87,6 +87,223 @@ public function testGetAdSection()
+            ->addImpression('http://ad.server.com/impression');
+
+        // create creative for ad section
+        $ad1
+            ->createLinearCreative()
+            ->setDuration(128)
+            ->setVideoClicksClickThrough('http://entertainmentserver.com/landing')
+            ->addVideoClicksClickTracking('http://ad.server.com/videoclicks/clicktracking')
+            ->addVideoClicksCustomClick('http://ad.server.com/videoclicks/customclick')
+            ->addTrackingEvent('start', 'http://ad.server.com/trackingevent/start')
+            ->addTrackingEvent('pause', 'http://ad.server.com/trackingevent/stop')
+            ->skipAfter(1519203721)
+            ->createMediaFile()
+                ->setDelivery('invalid_delivery')
+                ->setType('video/mp4')
+                ->setHeight(100)
+                ->setWidth(100)
+                ->setUrl('http://server.com/media.mp4');
+    }
+
+    /**
+     * Test for creating linear creative with adding extension
+     */
+    public function testCreateAdSectionWithAddingExtension()
+    {
+        $factory = new Factory();
+        $document = $factory->create('2.0');
+
+        // insert Ad section
+        $ad1 = $document
+            ->createInLineAdSection()
+            ->setId('ad1')
+            ->setAdSystem('Ad Server Name')
+            ->setAdTitle('Ad Title')
+            ->addImpression('http://ad.server.com/impression');
+
+        // create creative for ad section
+        $ad1
+            ->addExtension('extension_type', 'extension_value')
+            ->createLinearCreative()
+            ->setDuration(128)
+            ->setVideoClicksClickThrough('http://entertainmentserver.com/landing')
+            ->addVideoClicksClickTracking('http://ad.server.com/videoclicks/clicktracking')
+            ->addVideoClicksCustomClick('http://ad.server.com/videoclicks/customclick')
+            ->addTrackingEvent('start', 'http://ad.server.com/trackingevent/start')
+            ->addTrackingEvent('pause', 'http://ad.server.com/trackingevent/stop')
+            ->skipAfter(1519203721)
+            ->createMediaFile()
+                ->setType('video/mp4')
+                ->setHeight(100)
+                ->setWidth(100)
+                ->setUrl('http://server.com/media.mp4');
+
+        $adSections = $document->getAdSections();
+
+        $this->assertCount(1, $adSections);
+    }
+
+    /**
+     * Test for Document with set sequence
+     */
+    public function testCreateAdSectionWithSettingSequence()
+    {
+        $factory = new Factory();
+        $document = $factory->create('2.0');
+
+        // insert Ad section
+        $ad1 = $document
+            ->createInLineAdSection()
+            ->setId('ad1')
+            ->setAdSystem('Ad Server Name')
+            ->setAdTitle('Ad Title')
+            ->setSequence(0)
+            ->addImpression('http://ad.server.com/impression');
+
+        $this->assertSame('0', $ad1->getSequence());
```

Comment:
i think method getSequence should return int

Reply:
Unfortunately, the DOMElement::getAttribute will return string type.
Here is the error output:

```
PHPUnit 6.5.6 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.1.13 with Xdebug 2.6.0
Configuration: /home/lee/php-vast/tests/phpunit.xml

.......F

Time: 61 ms, Memory: 4.00MB

There was 1 failure:

1) Sokil\Vast\DocumentTest::testCreateAdSectionWithSettingSequence
Failed asserting that '0' is identical to 0.

/home/lee/php-vast/tests/DocumentTest.php:304

```

Reply:
so cast it to int?